packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -0,0 +1,82 @@
+import * as util from '../util';
+
+interface Options {
+  'ts-ignore'?: boolean;
+  'ts-nocheck'?: boolean;
+  'ts-check'?: boolean;
+}
+
+const defaultOptions: [Options] = [
+  {
+    'ts-ignore': true,
+    'ts-nocheck': true,
+    'ts-check': false,
+  },
+];
+
+type MessageIds = 'tsDirectiveComment';
+
+export default util.createRule<[Options], MessageIds>({
+  name: 'ban-ts-comment',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Bans “// @ts-<directive>” comments from being used',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    messages: {
+      tsDirectiveComment:
+        'Do not use "// @ts-{directive}" because it alters compilation errors.',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          'ts-ignore': {
+            type: 'boolean',
+            default: true,
+          },
+          'ts-nocheck': {
+            type: 'boolean',
+            default: true,
+          },
+          'ts-check': {
+            type: 'boolean',
+            default: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions,
+  create(context, [options]) {
+    const tsCommentRegExp = /^\/*\s*@ts-(ignore|check|nocheck)/;
```

Comment:
Ideally I'd use a named capture, but TypeScript doesn't transpile them, and babel isn't in use so 🤷‍♂

Reply:
can you add comment with TODO for it (prefferably with example)

we are planning to drop node 8 with next major release 
https://node.green/#ES2018-features

Reply:
Tbh I don't think the gain is worth it - they're cool, but TypeScript can't extract them, so I think it's a very small gain for something that is completely breaking for Node 8.

I'm happy to leave a todo comment :)